src/test/java/reactor/core/publisher/MonoSequenceEqualTest.java

```diff
@@ -15,11 +15,211 @@
  */
 package reactor.core.publisher;
 
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.atomic.LongAdder;
+
+import org.junit.Assert;
 import org.junit.Test;
+import org.reactivestreams.Subscription;
+import reactor.core.scheduler.Schedulers;
+import reactor.test.StepVerifier;
+import reactor.util.concurrent.QueueSupplier;
 
 public class MonoSequenceEqualTest {
 
 	@Test
-	public void normal() {
+	public void sequenceEquals() {
+		StepVerifier.create(Mono.sequenceEqual(
+						Flux.just("one", "two", "three"),
+						Flux.just("one", "two", "three")))
+		            .expectNext(Boolean.TRUE)
+		            .verifyComplete();
+	}
+
+	@Test
+	public void sequenceLongerLeft() {
+		StepVerifier.create(Mono.sequenceEqual(
+						Flux.just("one", "two", "three", "four"),
+						Flux.just("one", "two", "three")))
+		            .expectNext(Boolean.FALSE)
+		            .verifyComplete();
+	}
+
+	@Test
+	public void sequenceLongerRight() {
+		StepVerifier.create(Mono.sequenceEqual(
+						Flux.just("one", "two", "three"),
+						Flux.just("one", "two", "three", "four")))
+		            .expectNext(Boolean.FALSE)
+		            .verifyComplete();
+	}
+
+	@Test
+	public void sequenceErrorsLeft() {
+		StepVerifier.create(Mono.sequenceEqual(
+						Flux.just("one", "two").concatWith(Mono.error(new IllegalStateException())),
+						Flux.just("one", "two", "three")))
+		            .verifyError(IllegalStateException.class);
+	}
+
+	@Test
+	public void sequenceErrorsRight() {
+		StepVerifier.create(Mono.sequenceEqual(
+						Flux.just("one", "two", "three"),
+						Flux.just("one", "two").concatWith(Mono.error(new IllegalStateException()))))
+		            .verifyError(IllegalStateException.class);
+	}
+
+	@Test
+	public void sequenceErrorsBothPropagatesLeftError() {
+		StepVerifier.create(Mono.sequenceEqual(
+						Flux.just("one", "two", "three").concatWith(Mono.error(new IllegalArgumentException("left"))).hide(),
+						Flux.just("one", "two").concatWith(Mono.error(new IllegalArgumentException("right"))).hide()))
+		            .verifyErrorMessage("left");
```

Comment:
Actually, the publisher are subscribed one after the other, and they are requested by the `bufferSize` on subscription, which is 256 by default. The buffer request leads to the first publisher in the test to be immediately drained (only 3 elements) and the error encountered early and propagated. It can be inverted if the bufferSize is made smaller (like 1), because then in the parent drain loop there is also a `request(1)` performed on both publishers, left then right.... But!

While testing quickly with such a small initial request, I noticed that the left message was still propagated. When I made the left source larger (4 elements vs 2 elements in the right source), the right message finally got propagated... that's because at 1 element difference there is still a race between the 2 sources where the left array is at its 3rd value, right is still at 2nd value, and then left is completed and propagates an error while right is still emitting its 2nd value. What follows is that right is concurrently cancelled by the left error being propagated. With a larger difference, the small additional delay is enough for right to complete and propagate its own error.

Anyways, which error gets propagated is heavily dependent on the whole chain and its internals, so there shouldn't be any assumption here (in other words, the `sequenceEqual` doesn't make any guarantee as to which error it will propagate if the 2 sequences compared both terminate with an error, just that it will propagate an error)